### PR TITLE
Improve logical port call logic.

### PR DESCRIPTION
**Pull request by Levitan**
- Add field "Logical port type"

Closes: #2461

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/2468)

### CODE REVIEW

File: services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/client/GrpcSession.java [hunk 1]
```diff
@@ -127,9 +128,13 @@ public void shutdown() {
         Objects.requireNonNull(port.getLogicalPortNumber(), "Logical port number must not be null");
         Objects.requireNonNull(port.getPortNumbers(), "Port number must not be null");
 
+        if (port.getType() == null || port.getType().getNumber() == 0) {
```
Comment (timofei-durakov) [inline services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/client/GrpcSession.java]: magic number

File: services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/messaging/KafkaMessageListener.java [hunk 2]
```diff
@@ -41,7 +43,25 @@
...
+        messageProcessor.processRequest(message);
+    }
+
+    private boolean isValid(Message message) {
+        if (StringUtils.isEmpty(message.getCorrelationId())) {
+            log.warn("Received message without correlation id: {}", message);
+            return false;
+        }
+
+        //if (!(message instanceof InfoMessage) && !(message instanceof ErrorMessage)) {
```
Comment (timofei-durakov) [inline services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/messaging/KafkaMessageListener.java]: what is that?

File: services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/mapper/NoviflowResponseMapper.java [hunk 3]
```diff
@@ -27,11 +28,12 @@
...
 import org.mapstruct.Mapping;
 
-@Mapper(componentModel = "spring")
+@Mapper(componentModel = "spring", imports = {LogicalPortType.class})
 public interface NoviflowResponseMapper {
 
     @Mapping(source = "logicalportno", target = "logicalPortNumber")
     @Mapping(source = "portnoList", target = "portNumbers")
+    @Mapping(target = "type", expression = "java(LogicalPortType.forNumber(port.getLogicalporttypeValue()))")
     org.openkilda.messaging.model.grpc.LogicalPort toLogicalPort(LogicalPort port);
```
Comment (timofei-durakov) [inline services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/mapper/NoviflowResponseMapper.java]: I would prefer these methods to be named map

File: services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/messaging/KafkaMessageProducer.java [hunk 4]
```diff
@@ -32,14 +32,14 @@
 public class KafkaMessageProducer {
 
     @Autowired
-    private KafkaTemplate<String, InfoData> kafkaTemplate;
+    private KafkaTemplate<String, Message> kafkaTemplate;
 
     /**
      * Sends message to kafka topic.
      */
     // TODO seems like message should be Message subclass
```
Comment (timofei-durakov) [inline services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/messaging/KafkaMessageProducer.java]: what's about TODO?

File: services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/messaging/MessageProcessor.java [hunk 5]
```diff
@@ -46,29 +52,56 @@
...
+                    });
 
         }
     }
 
-    private void sendResponse(InfoData message) {
+    private void sendResponse(InfoData data, String correlationId) {
         // TODO topic hardcode
+        InfoMessage message = new InfoMessage(data, System.currentTimeMillis(), correlationId);
         messageProducer.send("grpc.response", message);
```
Comment (timofei-durakov) [inline services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/messaging/MessageProcessor.java]: address this todo

File: services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/messaging/MessageProcessor.java [hunk 6]
```diff
@@ -46,29 +52,56 @@
     /**
      * Process request.
      *
-     * @param command a command data.
+     * @param message a command data.
      */
-    public void processRequest(CommandData command) {
+    public void processRequest(CommandMessage message) {
+        CommandData command = message.getData();
         if (command instanceof CreateLogicalPortRequest) {
```
Comment (timofei-durakov) [inline services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/messaging/MessageProcessor.java]: create private methods for each if statement in this case

File: services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/client/GrpcSession.java [hunk 7]
```diff
@@ -126,10 +130,12 @@ public void shutdown() {
     public CompletableFuture<List<CliReply>> setLogicalPort(LogicalPortDto port) {
         Objects.requireNonNull(port.getLogicalPortNumber(), "Logical port number must not be null");
```
Comment (timofei-durakov) [inline services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/client/GrpcSession.java]: what if logical port itself is null

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/grpc/DeleteLogicalPortRequest.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import lombok.Data;
+
+@Data
+public class DeleteLogicalPortRequest extends CommandData {
+
+    @JsonProperty("address")
+    private String address;
+
+    @JsonProperty("logical_port_number")
+    private Integer logicalPortNumber;
```
Comment (timofei-durakov) [inline services/src/messaging/src/main/java/org/openkilda/messaging/command/grpc/DeleteLogicalPortRequest.java]: why not primitive?

File: services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/client/GrpcSession.java (hunk 7 above)
Comment (Levitan) [reply services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/client/GrpcSession.java]: This case will be handled, and return a correct error

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/grpc/DeleteLogicalPortRequest.java (hunk 8 above)
Comment (Levitan) [reply services/src/messaging/src/main/java/org/openkilda/messaging/command/grpc/DeleteLogicalPortRequest.java]: This is mistake. Fixed in CreateRequest too

**Pull request closed without merging**